pkg/cmd/ssh/ssh_test.go

```diff
@@ -632,35 +659,23 @@ var _ = Describe("SSH Command", func() {
 		})
 	})
 
-	// TODO not clear how to pass this test will take a look in the future https://github.com/gardener/gardenctl-v2/issues/323
-	// Describe("ValidArgsFunction", func() {
-	// 	BeforeEach(func() {
-	// 		monitoringNode := &corev1.Node{
-	// 			ObjectMeta: metav1.ObjectMeta{
-	// 				Name: "monitoring",
-	// 			},
-	// 		}
-
-	// 		workerNode := &corev1.Node{
-	// 			ObjectMeta: metav1.ObjectMeta{
-	// 				Name: "worker",
-	// 			},
-	// 		}
-
-	// 		shootClient = internalfake.NewClientWithObjects(monitoringNode, workerNode)
-	// 	})
-
-	// 	It("should find nodes based on their prefix", func() {
-	// 		options := ssh.NewSSHOptions(streams)
-	// 		cmd := ssh.NewCmdSSH(factory, options)
-
-	// 		// let the magic happen; should find "monitoring" node based on this prefix
-	// 		suggestions, directive := cmd.ValidArgsFunction(cmd, nil, "mon")
-	// 		Expect(directive).To(Equal(cobra.ShellCompDirectiveNoFileComp))
-	// 		Expect(suggestions).To(HaveLen(1))
-	// 		Expect(suggestions).To(Equal([]string{"monitoring"}))
-	// 	})
-	// })
+	Describe("ValidArgsFunction", func() {
+		BeforeEach(func() {
+			seedClient = internalfake.NewClientWithObjects(testMachine)
```

Comment:
Hi @tedteng, the previous test had two nodes:
- `monitorning`
- `worker`

Valid args was called with `mon` so that the suggestions returned `monitoring`

Now you have changed the test to only register (`internalfake.NewClientWithObjects(testMachine)`) one machine object. What happened to the test with the two nodes?

Instead (or additionally) have a test with two items of each:
- nodes:
  - `monitoring1`
  - `worker1`
- machines:
  - `monitoring2`
  - `worker2`

call cmd.ValidArgsFunction(cmd, nil, "mon") which should return `monitoring1` and `monitoring2`

Reply:
I changed a little bit in here
Machine1(name:monitoring1) => Node(name:monitoring1-node)
Machine2(name:monitoring2) => Node(name:monitoring2-node) assume This node is not ready(not joined)
then call cmd.ValidArgsFunction(cmd, nil, "mon") which should return monitoring1 and monitoring2
16456f1